### PR TITLE
jewel: rgw: radosgw-admin: add an option to reset user stats

### PR DESCRIPTION
**Pull request by theanalyst**
http://tracker.ceph.com/issues/23338 & http://tracker.ceph.com/issues/23721

### CODE REVIEW

**Comment by smithfarm** [comment]
@theanalyst Do Not Merge until https://github.com/ceph/ceph/pull/20877 is merged, right?

**Comment by theanalyst** [comment]
@smithfarm right

**Comment by smithfarm** [comment]
@theanalyst The master PR #20853 has a third commit "rgw: rgw-rados, rgw-admin add an option to recalculate user stats" which was not cherry-picked. Was that by design?

**Comment by smithfarm** [comment]
@theanalyst ACK, thanks!

**Comment by theanalyst** [comment]
passes local testing

File: src/cls/user/cls_user.cc [hunk 1]
```diff
@@ -370,6 +371,54 @@ static int cls_user_get_header(cls_method_context_t hctx, bufferlist *in, buffer
...
+	auto bliter = bl.begin();
+	decode(e, bliter);
+      } catch (buffer::error& err) {
+	CLS_LOG(0, "ERROR: failed to decode bucket entry for %s", kv.first.c_str());
+	return -EIO;
+      }
+      from_index = kv.first;
+      add_header_stats(&header.stats, e);
+    }
+  } while (rc == MAX_ENTRIES);
```
Comment (oritwas) [inline src/cls/user/cls_user.cc]: I don't think this will work, try:
while (keys.size() == MAX_ENTRIES)

**Comment by yuriw** [comment]
[wip-yuri4-testing-2018-05-23-1713-jewel](https://trello.com/c/sEAU8Gda/317-wip-yuri4-testing-2018-05-23-1713-jewel)

**Pull request merged**